test/rule-matches/label-content-name-mismatch-matches.js

```diff
@@ -0,0 +1,144 @@
+describe('label-content-name-mismatch-matches tests', function() {
+	'use strict';
+
+	var fixture = document.getElementById('fixture');
+	var queryFixture = axe.testUtils.queryFixture;
+	var rule = axe._audit.rules.find(function(rule) {
+		return rule.id === 'label-content-name-mismatch';
+	});
+
+	afterEach(function() {
+		fixture.innerHTML = '';
+		axe._tree = undefined;
+	});
+
+	it('returns false if given element has no role', function() {
+		var vNode = queryFixture('<div id="target"></div>');
+		var actual = rule.matches(vNode.actualNode, vNode);
+		assert.isFalse(actual);
+	});
+
+	it('returns false if element role is not supported with name from contents', function() {
+		var vNode = queryFixture('<textarea id="target"></textarea>');
```

Comment:
This elm could just as easily be false because it doesn't have an accessible name and/or text content. This test doesn't prove what you say it does.